src/main/java/fi/dy/masa/servux/dataproviders/StructureDataProvider.java

```diff
@@ -1,94 +1,156 @@
     protected final NbtCompound metadata = new NbtCompound();
     protected int timeout = 30 * 20;
     protected int updateInterval = 40;
     protected int retainDistance;
 
+    // FIXME --> Move out of structures channel in the future
+    private BlockPos spawnPos = BlockPos.ORIGIN;
+    private int spawnChunkRadius = -1;
+    private boolean refreshSpawnMetadata;
+
     protected StructureDataProvider()
     {
         super("structure_bounding_boxes",
-              StructureDataPacketHandler.CHANNEL, StructureDataPacketHandler.PROTOCOL_VERSION,
-              "Structure Bounding Boxes data for structures such as Witch Huts, Ocean Monuments, Nether Fortresses etc.");
-
-        this.metadata.putString("id", StructureDataPacketHandler.CHANNEL.toString());
+                ServuxStructuresHandler.CHANNEL_ID,
+                ServuxStructuresHandler.PROTOCOL_VERSION,
+    "Structure Bounding Boxes data for structures such as Witch Huts, Ocean Monuments, Nether Fortresses etc.");
+
+        this.metadata.putString("name", this.getName());
+        this.metadata.putString("id", this.getNetworkChannel().toString());
+        this.metadata.putInt("version", this.getProtocolVersion());
+        this.metadata.putString("servux", Reference.MOD_STRING);
         this.metadata.putInt("timeout", this.timeout);
-        this.metadata.putInt("version", StructureDataPacketHandler.PROTOCOL_VERSION);
+
+        // TODO --> Move out of structures channel in the future
+        this.metadata.putInt("spawnPosX", this.getSpawnPos().getX());
+        this.metadata.putInt("spawnPosY", this.getSpawnPos().getY());
+        this.metadata.putInt("spawnPosZ", this.getSpawnPos().getZ());
+        this.metadata.putInt("spawnChunkRadius", this.getSpawnChunkRadius());
+    }
+
+    @Override
+    public void setEnabled(boolean toggle)
+    {
+        this.enabled = toggle;
+    }
+
+    @Override
+    public void registerHandler()
+    {
+        ServerPlayHandler.getInstance().registerServerPlayHandler(HANDLER);
+    }
+
+    @Override
+    public void unregisterHandler()
+    {
+        ServerPlayHandler.getInstance().unregisterServerPlayHandler(HANDLER);
     }
 
     @Override
     public boolean shouldTick()
     {
-        return true;
+        return this.enabled;
     }
 
     @Override
-    public IPluginChannelHandler getPacketHandler()
+    public IPluginServerPlayHandler<ServuxStructuresPayload> getPacketHandler()
     {
-        return StructureDataPacketHandler.INSTANCE;
+        return HANDLER;
     }
 
     @Override
     public void tick(MinecraftServer server, int tickCounter)
     {
         if ((tickCounter % this.updateInterval) == 0)
         {
-            if (this.registeredPlayers.isEmpty() == false)
+            //Servux.printDebug("=======================\n");
+            //Servux.printDebug("tick: %d - %s\n", tickCounter, this.isEnabled());
+
+            List<ServerPlayerEntity> playerList = server.getPlayerManager().getPlayerList();
```

Comment:
Iterates through PlayerManager's list, because we might have MiniHUD clients who have their Structures Disabled.